src/qibo/gates/abstract.py

```diff
@@ -391,7 +391,7 @@ def apply_density_matrix(self, backend, state, nqubits):
         return backend.apply_gate_density_matrix(self, state, nqubits)
 
     def apply_clifford(self, backend, state, nqubits):
-        return backend.apply_gate_clifford(self, state, nqubits)
+        return backend.apply_gate_clifford(self, state[:-1], nqubits)
```

Comment:
Are you overriding this for the measurement gate?

I would have said that it should have been cut where you can distinguish measurement from everything else.

Reply:
Yes the `M` gates overrides this and passes the complete matrix instead.